File: test/data/phenology/crop2ml/algo/r/UpdateLeafFlag.r

if (phase >= 1 and phase< 4):
    if (leafNumber > 0):
        if (hasFlagLeafLiguleAppeared == 0 and (finalLeafNumber > 0 and leafNumber >= finalLeafNumber)):
            hasFlagLeafLiguleAppeared = 1
            if  "FlagLeafLiguleJustVisible" not in calendarMoments:
                calendarMoments.append("FlagLeafLiguleJustVisible")
                calendarCumuls.append(cumulTT)
                calendarDates.append(currentdate)
    else:
        hasFlagLeafLiguleAppeared = 0


    

File: test/data/phenology/crop2ml/algo/cs/UpdateLeafFlag.cs
if (phase >= 1.0 && phase< 4.0)
{
    if (leafNumber > 0.0)
    {
        if (hasFlagLeafLiguleAppeared == 0 && (finalLeafNumber > 0.0 && leafNumber >= finalLeafNumber))
        {
            hasFlagLeafLiguleAppeared = 1;
            if  (calendarMoments.Contains("FlagLeafLiguleJustVisible")==false)
            {
                calendarMoments.Add("FlagLeafLiguleJustVisible");
                calendarCumuls.Add(cumulTT);
                calendarDates.Add(currentdate);
            }
        }
    }
    else
        hasFlagLeafLiguleAppeared = 0;
}


    
